heudiconv/convert.py

```diff
@@ -611,6 +619,20 @@ def save_converted_files(res, item_dicoms, bids_options, outtype, prefix, outnam
                             imgtype, "_echo-%d%s" % (echo_number, imgtype)
                         )
                         break
+            # Now check if this run is multi-orientation (v-nav or localizer)
+            if bids_meta and is_multiorient and 'acq-' not in this_prefix_basename:
+                iop = bids_meta.get('ImageOrientationPatientDICOM')
+                iop = [round(x) for x in iop]
+                cross_prod = [
+                    iop[1]*iop[5]-iop[2]*iop[4],
+                    iop[2]*iop[3]-iop[0]*iop[5],
+                    iop[0]*iop[4]-iop[1]*iop[3]]
+                cross_prod = [abs(x) for x in cross_prod]
+                slice_orient = ['saggital','coronal','axial'][cross_prod.index(1)]
+                bids_pairs = this_prefix_basename.split('_')
+                ses_or_sub_idx = sum([bids_pair.split('-')[0] in ['sub','ses'] for bids_pair in bids_pairs])
+                bids_pairs.insert(ses_or_sub_idx,'acq-%s'%slice_orient)
+                this_prefix_basename = '_'.join(bids_pairs)
```

Comment:
not immediately clear to me why it is not just adding `_acq-` at the end and goes through all splitting and then joining.  Please
- add a brief comment what is about to happen (something per sub/session?) and why
- make it more pep-8 friendly here ( add spaces after `,` between list entries and function arguments)

re "at the end" -- you added this code AFTER possible addition of suffixes... feels like it should be before, and not interfere with suffixes handling (there is also condition after it which would not add any suffix if any changes to this_prefix_basename happen)

Reply:
and altogether - it might be worth moving this code block into a function (e.g. , so it could be unittested on its own and also make code not so long right here

Reply:
Agreed on the separate function following #424 example. Will clean up.
It thought that the BIDS key-value pairs are supposed to be in a specific order, and that `_acq-` is supposed to go after `sub-*[_ses-*]` .
As the filenames prefixes are already formed, I found this to be the easiest way to insert it at the proper location. For the other cases that #424 is dealing with (eg. https://github.com/nipy/heudiconv/pull/424/files#diff-a48009f9efe6cc64ffdb74d06e831296R341) , they choose to search through potential key-value pairs that are supposed to be after the inserted key-value, here it does the reverse because there would be too many tags to search for, as it is supposed to be right after sub or ses. Does it make sense? 
Will add comment to make it more obvious.

Reply:
> It thought that the BIDS key-value pairs are supposed to be in a specific order, and that _acq- is supposed to go after sub-*[_ses-*] .

that is correct!  but it should go before suffix (like `_bold`) which comes before extension (`.nii.gz`).

Reply:
I used this file: https://github.com/bids-standard/pybids/blob/master/bids/layout/config/bids.json as a reference, which lists the following path.

```
"sub-{subject}[/ses-{session}]/{datatype<anat>|anat}/sub-{subject}[_ses-{session}][_acq-{acquisition}][_ce-{ceagent}][_rec-{reconstruction}]_{suffix<T1w|T2w|T1rho|T1map|T2map|T2star|FLAIR|FLASH|PDmap|PD|PDT2|inplaneT[12]|angio>}.{extension<nii|nii.gz|json>|nii.gz}"
```

Also in the BIDS specs: 
https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/01-magnetic-resonance-imaging-data.html#anatomy-imaging-data

I am confused, does the order of key-value pairs matter?

Reply:
yes! order matters, sorry -- parts of my comment were misleading and  incorrect.  The ultimate order prescribed in BIDS itself in the [entity table](https://bids-specification.readthedocs.io/en/stable/99-appendices/04-entity-table.html).  I think we need to add a function which places a new key in the correct position based on other keys present.
What I was pointing also that it matters for those `_key-value` pairs to come before `_{suffix<T1w...}.{extension<nii...}"`, thus my original comment: 

re "at the end" -- you added this code AFTER possible addition of suffixes... feels like it should be before, and not interfere with suffixes handling (there is also condition after it which would not add any suffix if any changes to this_prefix_basename happen)

Reply:
I admit that I pretty much adapted the existing cases (which are refactored in #424), so I am basically following the same renaming procedure (even if in their cases they str.replace to insert the tag). 
The variable name they used is `this_prefix_basename` it seems to be only the prefix, but I will be happy to adapt to any change to #424 if necessary.